src/background/services/keyAutoAdd.ts

```diff
@@ -137,15 +143,20 @@ export class KeyAutoAddService {
           ),
         );
       } else if (message.action === 'PROGRESS') {
-        // can also save progress to show in popup
-        const currentStep = this.getCurrentStep(message.payload.steps);
+        const steps = message.payload.steps;
+        const timeoutIn = steps
+          .filter(({ status }) => status === 'pending' || status === 'active')
+          .reduce((acc, { maxDuration }) => acc + maxDuration, BASE_TIMEOUT);
```

Comment:
Each step now has a default `maxDuration` (of 4 seconds). "Waiting for login" still has same `maxDuration` (10min), and each step can specify its own as needed.